block-storage/src/main/scala/coop/rchain/blockstorage/util/Crc32.scala

```diff
@@ -0,0 +1,37 @@
+package coop.rchain.blockstorage.util
+
+import java.nio.ByteBuffer
+import java.util.zip.CRC32
+
+import cats.Monad
+import cats.implicits._
+
+final class Crc32[F[_]: Monad](internal: CRC32) {
+  def update(bytes: Array[Byte]): F[Unit] =
+    internal.update(bytes).pure[F]
+
+  def update(byteBuffer: ByteBuffer): F[Unit] =
+    internal.update(byteBuffer).pure[F]
+
+  def value: F[Long] =
+    internal.getValue.pure[F]
+
+  def bytes: F[Array[Byte]] =
+    value.map { value =>
+      val byteBuffer = ByteBuffer.allocate(8)
+      byteBuffer.putLong(value)
+      byteBuffer.array()
+    }
+
+  def reset: F[Unit] =
+    internal.reset().pure[F]
```

Comment:
This should be `Sync[F].delay { … }` as `reset` is effectful. `.pure[F]` is still eager.